### PR TITLE
extensions: Add VK_KHR_get_surface_capabilities2

### PR DESCRIPTION
**Pull request by asnordstrand**

### CODE REVIEW

File: ash/src/extensions/khr/get_surface_capabilities2.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            .result_with_success(surface_capabilities)
+    }
+
+    #[doc = "https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetPhysicalDeviceSurfaceFormats2KHR.html"]
+    pub unsafe fn get_physical_device_surface_formats2(
+        &self,
+        physical_device: vk::PhysicalDevice,
+        surface_info: &vk::PhysicalDeviceSurfaceInfo2KHR,
+    ) -> VkResult<Vec<vk::SurfaceFormat2KHR>> {
+        read_into_uninitialized_vector(|count, data| {
```
Comment (MarijnS95) [inline ash/src/extensions/khr/get_surface_capabilities2.rs:42]: `vk::SurfaceFormat2KHR` has an `s_type` and `p_next` (unlike `vk::SurfaceFormatKHR`) hence needs to be initialized with `read_into_defaulted_vector` here. Perhaps this `p_next` should be settable externally just like other call-sites that have a separate `_len()` function (see for example `fn get_device_image_sparse_memory_requirements` in `maintenance4`).

**Review by MarijnS95 — COMMENTED** [review]
Nice! Good use of `Default::default()` for `vk::SurfaceCapabilities2KHR`, but I think you need to do the same for `vk::SurfaceFormat2KHR`.

I don't think `rustfmt` is going to be happy with `fp()` and `name()` on a single line though?

File: ash/src/extensions/khr/get_surface_capabilities2.rs (hunk 1 above)
Comment (asnordstrand) [reply ash/src/extensions/khr/get_surface_capabilities2.rs]: Good points that I overlooked. Have made the suggested changes, let me know if I'm handling the errors correctly.

**Review by MarijnS95 — APPROVED** [review]
Looks totally right to me now!

**Comment by MarijnS95** [comment]
By the way, don't forget to add a note to `./Changelog.md`, see the other new extensions in there for examples.

(We should setup a PR template listing these necessary items later)

**Pull request merged**